File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/Slime.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class Slime extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 35;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 26;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 130;
        this.baseConstitution = 40;
        this.baseIntelligence = 2;
        this.baseDexterity = 6;
        this.baseDefense = 40;
        this.baseMagicDefense = 0;
        this.imageId = R.drawable.unit_slime;
        this.idName = R.string.enemy_slime_name;
        this.idDescription = R.string.enemy_slime_description;
        this.passiveSkill = Skills.PASSIVE_NONE;
        this.activeSkill = Skills.ACTIVE_NONE;
        this.expGiven = 24;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 1), 650);
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 2), 200);
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 3), 50);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/Slime.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class Slime : Enemy() {
    override fun getMaxDamage(): Int = 35
    override fun getMinDamage(): Int = 26
    override fun isMagic(): Boolean = false
    override fun isRanged(): Boolean = false

    override fun configureStatistics() {
        baseMaxHp = 130
        baseConstitution = 40
        baseIntelligence = 2
        baseDexterity = 6
        baseDefense = 40
        baseMagicDefense = 0
        imageId = R.drawable.unit_slime
        idName = R.string.enemy_slime_name
        idDescription = R.string.enemy_slime_description
        passiveSkill = Skills.PASSIVE_NONE
        activeSkill = Skills.ACTIVE_NONE
        expGiven = 24
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 1), 650)
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 2), 200)
        linkedHashMap.put(ItemWrapper.getInstance("GreenSlime", 3), 50)
        return linkedHashMap
    }
}
